client/jetpack-cloud/components/user-feedback-modal-form/index.tsx

```diff
@@ -19,10 +22,32 @@ const DEFAULT_RATING_VALUE = 0;
 
 export default function UserFeedbackModalForm( { show, onClose }: Props ) {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
 
 	const [ feedback, setFeedback ] = useState( DEFAULT_FEEDBACK_VALUE );
 	const [ rating, setRating ] = useState( DEFAULT_RATING_VALUE );
 
+	const { isSubmittingFeedback, submitFeedback, isSubmissionSuccessful } =
+		useSubmitProductFeedback();
+
+	const onModalClose = useCallback( () => {
+		setFeedback( DEFAULT_FEEDBACK_VALUE );
+		setRating( DEFAULT_RATING_VALUE );
+		onClose?.();
+	}, [ onClose ] );
+
+	useEffect( () => {
+		if ( isSubmissionSuccessful ) {
+			dispatch(
+				successNotice( translate( 'Thank you for your feedback!' ), {
+					id: 'submit-product-feedback-success',
+					duration: 5000,
+				} )
+			);
+			onModalClose?.();
```

Comment:
[non-blocker] `onModalClose` here is always defined, so we can omit `?.()`.

Reply:
🤦

Reply:
Fixed: https://github.com/Automattic/wp-calypso/compare/dd8d625cbc3f11a3ef193432a20a6ef020123332..c6ac6b03c79e45dc211657e4eb70a20af30ab973